### PR TITLE
Egg Dropping problem in Java 

### LINKED ISSUE #5144: Egg Dropping Puzzle
## 💥 Proposal

I want to implement the Egg Dropping Puzzle problem in Java

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes!

### PR DESCRIPTION
**Pull request by vi-hna-ja**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Imagine there is a building of N floors and you have K eggs.
How many minimum number of attempts it would take to find the minimum floor (or first floor) F (0 <= f <= N)  where any egg dropped at a floor higher than f will break, and any egg dropped at or below the floor f will not break.

## Related Issues or Pull Requests

Fixes: #5144

### CODE REVIEW

**Comment by vi-hna-ja** [comment]
Kindly review my PR @HarshCasper

File: Java/dp/EggDroppingProblem.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    
+Output: 
+The critical floor is: 4
+    
+Time Complexity: O(K * log(N)) where K = eggs we have and N = building height we are on.
+Space Complexity: O(K)
+
+*/ 
+
+
```
Comment (ankitaggarwal23) [inline Java/dp/EggDroppingProblem.java]: Remove the extra one line

File: Java/dp/EggDroppingProblem.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            for(int i = numEggs; i > 0; i--) {
+                cache[attempts][i] = 1 + cache[attempts - 1][i] + cache[attempts - 1][i - 1];
+            }
+        }
+
+        return attempts;
+    }
+}
+
+
```
Comment (ankitaggarwal23) [inline Java/dp/EggDroppingProblem.java]: Remove the extra one line

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: Java/dp/EggDroppingProblem.java (hunk 2 above)
Comment (vi-hna-ja) [reply Java/dp/EggDroppingProblem.java]: Removed.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**